cxplugin-agent/src/main/java/com/checkmarx/teamcity/agent/CxBuildProcess.java

```diff
@@ -153,22 +150,16 @@ public BuildFinishedStatus call() throws Exception {
             //assert if expected exception is thrown  OR when vulnerabilities under threshold OR when policy violated
             //String buildFailureResult = ShragaUtils.getBuildFailureResult(config, ret.getSastResults(), ret.getOsaResults());
             ScanSummary scanSummary = new ScanSummary(config, ret.getSastResults(), ret.getOsaResults(), ret.getScaResults());
-            checkExceptions(ret);
-            if (scanSummary.hasErrors() || sastWaitEx != null || sastCreateEx != null ||
-                    osaCreateEx != null || osaWaitEx != null ||
-                    scaCreateEx != null || scaWaitEx != null) {
+            if (scanSummary.hasErrors() || ret.getGeneralException() != null ||
+                    (config.isSastEnabled() && (ret.getSastResults() == null || ret.getSastResults().getException() != null)) ||
+                    (config.isOsaEnabled() && (ret.getOsaResults() == null || ret.getOsaResults().getException() != null)) ||
+                    (config.isAstScaEnabled() && (ret.getScaResults() == null || ret.getScaResults().getException() != null))) {
                 printScanBuildFailure(scanSummary, ret, logger);
                 return BuildFinishedStatus.FINISHED_FAILED;
             }
             return BuildFinishedStatus.FINISHED_SUCCESS;
         } catch (InterruptedException e) {
             logger.error("Interrupted exception: " + e.getMessage());
-
-            //TODO CHECK HOW TO CANCLE SCAN IN NEW COMMON
```

Comment:
cancel is performed using CxClientDelegator in Jenkins. You can test it by explicitly calling this function from a Test code while the scan is in-progress.  If you see it is working, remove the TODO and uncomment the cancelScan function on like 184

Reply:
These are the removed lines. Have removed the TODO and unnecessary commented lines.

Reply:
ok. My bad